presto-main/src/main/java/com/facebook/presto/sql/planner/SubqueryPlanner.java

```diff
@@ -291,6 +292,33 @@ private PlanBuilder appendQuantifiedComparisonApplyNode(PlanBuilder subPlan, Qua
             // given subquery is already appended
             return subPlan;
         }
+        switch (quantifiedComparison.getComparisonType()) {
+            case EQUAL:
+                switch (quantifiedComparison.getQuantifier()) {
+                    case ANY:
+                    case SOME:
+                        // A = ANY B <=> A IN B
+                        InPredicate inPredicate = new InPredicate(quantifiedComparison.getValue(), quantifiedComparison.getSubquery());
+                        subPlan.getTranslations().addIntermediateMapping(quantifiedComparison, inPredicate);
+                        return appendInPredicateApplyNode(subPlan, inPredicate, correlationAllowed);
```

Comment:
Can you please provide more details on your suggestion? If I understood correctly, you suggest rewriting "=ANY" to "IN" in AstBuilder, so that there's no need to deal with it later. If so, I though about this approach before and didn't like it. First of all, AST is close a representation of nodes of an actual query, so doing structural rewrites there seems like not a good idea. Next, the logic will be scattered across AST rewrite and plan generation, unlike now when it's all in a planner.